### PR TITLE
Fix incorrect computation of affine shear matrix around center point

### PR DESCRIPTION
**Pull request by Vihtoriaaa**
#### Changes

- An incorrect mathematical calculation of the affine shear matrix around center point was fixed, since `m23` was missing a negative sign when calculating `shear_mat`. 
- The `get_shear_matrix2d` test has been updated to work correctly due to the changes made.  

Fixes [#1698](https://github.com/kornia/kornia/issues/1698)

#### Type of change

- [x] 📚  Documentation Update
- [x] 🧪 Tests Cases
- [x] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📝 This change requires a documentation update

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Comment by Vihtoriaaa** [comment]
@johnnv1 Hi! Can you please review my PR if you have time to check whether everything is ok and it can me merged?:) thanks!

File: kornia/geometry/transform/imgwarp.py [hunk 1]
```diff
@@ -660,9 +660,9 @@ def get_shear_matrix2d(center: Tensor, sx: Optional[Tensor] = None, sy: Optional
     sx_tan = torch.tan(sx)
     sy_tan = torch.tan(sy)
     ones = torch.ones_like(sx)
-    shear_mat = stack([ones, -sx_tan, sx_tan * y, -sy_tan, ones + sx_tan * sy_tan, sy_tan * (sx_tan * y + x)], -1).view(
-        -1, 2, 3
-    )
+    shear_mat = stack(
+        [ones, -sx_tan, sx_tan * y, -sy_tan, ones + sx_tan * sy_tan, sy_tan * (x - sx_tan * y)], dim=-1
```
Comment (edgarriba) [inline kornia/geometry/transform/imgwarp.py:664]: if this math is incorrect -- can you add an additional unit test showing the fix? /cc @ducha-aiki

**Review by edgarriba — CHANGES_REQUESTED** [review]

File: kornia/geometry/transform/imgwarp.py (hunk 1 above)
Comment (Vihtoriaaa) [reply kornia/geometry/transform/imgwarp.py:664]: @edgarriba Hi! I am not quite sure how to show that the bug is fixed with additional unit test - can you please give me a hint regarding that? Should I compare the 'old math' and 'new math' to see which one is the correct one? or what the best way would be to do this ? thanks:)

Comment (Vihtoriaaa) [reply kornia/geometry/transform/imgwarp.py:664]: so would it be ok if I test whether expected (controlled matrix with numbers) is the same as the matrix that is computed with changed get_shear_matrix2d function?

Comment (edgarriba) [reply kornia/geometry/transform/imgwarp.py:664]: that would work

Comment (Vihtoriaaa) [reply kornia/geometry/transform/imgwarp.py:664]: @edgarriba Added a unit test, let me know if it works :) thanks!

**Review by edgarriba — APPROVED** [review]

**Pull request merged**